services/consensusalgo/benchmarkconsensus/test/leader_test.go

```diff
@@ -6,11 +6,121 @@ import (
 	"testing"
 )
 
-func TestLeaderProposesBlocks(t *testing.T) {
+func newLeaderHarnessAndInit(t *testing.T, ctx context.Context) *harness {
+	h := newHarness(true)
+	h.expectNewBlockProposalNotRequested()
+	h.expectCommitSent(0, h.config.NodePublicKey())
+	h.createService(ctx)
+	h.verifyCommitSent(t)
+	return h
+}
+
+func TestLeaderInit(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		h := newLeaderHarnessAndInit(t, ctx)
+		h.verifyNewBlockProposalNotRequested(t)
+		h.verifyHandlerRegistrations(t)
+	})
+}
+
+func TestLeaderCommitsNextBlockAfterEnoughConfirmations(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		h := newLeaderHarnessAndInit(t, ctx)
+
+		h.expectNewBlockProposalRequestedAndSaved(1)
+		h.expectCommitSent(1, h.config.NodePublicKey())
+		h.receivedCommittedViaGossipFromSeveral(3, 0, true, true)
+		h.verifyNewBlockProposalRequestedAndSaved(t)
+		h.verifyCommitSent(t)
+
+		h.expectNewBlockProposalRequestedAndSaved(2)
+		h.expectCommitSent(2, h.config.NodePublicKey())
+		h.receivedCommittedViaGossipFromSeveral(3, 1, true, true)
+		h.verifyNewBlockProposalRequestedAndSaved(t)
+		h.verifyCommitSent(t)
+	})
+}
+
+func TestLeaderRetriesCommitOnError(t *testing.T) {
```

Comment:
how does this this test assert that the leader retries? I don't see anything related to retries in the code